plugins/repository-azure/src/test/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceTests.java

```diff
@@ -392,8 +392,9 @@ public void testGetDefaultClientWithPrimaryAndSecondaries() {
     @Deprecated
     public void testGetSelectedClientDefault() {
```

Comment:
Shouldn't this test still be calling getSelectedClient (which in turn calls generateOperationContext)? Otherwise the name should be updated.

Reply:
No. Because `getSelectedClient` never calls `generateOperationContext` where the NPE happened.
I'm going to rename the method indeed to make that more obvious.